lib/types/src/partial_sum_map.rs

```diff
@@ -58,7 +55,14 @@ impl<K: seal::PartialSumKey, V> PartialSumMap<K, V> {
     ///
     /// `O(1)`
     pub fn max_index(&self) -> Option<K> {
-        self.keys.last().copied()
+        self.keys.last().cloned()
+    }
+
+    /// Get the current (virtual) size of this map. This is the sum of all `count` arguments passed to `push` until now.
+    ///
+    /// `O(1)`
+    pub fn size(&self) -> K {
```

Comment:
I’m torn on the `K` return type here. Intuitively I’d expect this to return `usize` like the `len` functions for other datastructures do, even if this one isn’t really limited to `usize` elements. So probably:

```suggestion
    pub fn len(&self) -> usize {
```

Reply:
I've just pushed a commit documenting why I'm not returning usize here: basically, if K is a BigInt type, it could go overboard just by having a few really large intervals of BigInts.

And that's also the reason why I didn't pick len(), because len() is usually used with an usize return, and so I thought it'd be confusing. That said I feel much less strongly about this one, WDYT?